src/sssom/io.py

```diff
@@ -130,8 +130,22 @@ def split_file(input_path: str, output_directory: Union[str, Path]) -> None:
     write_tables(splitted, output_directory)
 
 
+@deprecated(
+    deprecated_in="0.4.3",
+    details="This functionality for loading SSSOM metadata from a YAML file is deprecated from the "
+    "public API since it has internal assumptions which are usually not valid for downstream users.",
+)
 def get_metadata_and_prefix_map(
     metadata_path: Union[None, str, Path] = None, *, prefix_map_mode: Optional[MergeMode] = None
+) -> Tuple[Converter, MetadataType]:
+    """Load metadata and a prefix map in a deprecated way."""
+    return _get_metadata_and_prefix_map(
+        metadata_path=metadata_path, prefix_map_mode=prefix_map_mode
+    )
+
+
+def _get_metadata_and_prefix_map(
```

Comment:
I don't know if it's too much to handle in this PR as well, but per this [discussion thread](https://github.com/mapping-commons/sssom-py/pull/485#discussion_r1452770502), is this also a good time to rename this function to `_get_converter_and_metadata()`?

Reply:
Yes I agree @joeflack4. Now or never! Can we rename?

Reply:
Done in be8d768